src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -443,4 +514,74 @@ public enum Directive
         /// </summary>
         Stop,
     }
+
+    public static class DirectiveExtensions
+    {
+        public static KeyValuePair<Type, Directive> When<TException>(this Directive self) where TException : Exception
+        {
+            return new KeyValuePair<Type, Directive>(typeof(TException),self);
+        }
+    }
+
+    public interface IDecider
+    {
+        Directive Decide(Exception cause);        
+    }
+
+    public static class Decider
+    {
+        public static DeployableDecider From(Directive defaultDirective, params KeyValuePair<Type, Directive>[] pairs)
+        {
+            return new DeployableDecider(defaultDirective,pairs);
+        }
+
+        public static LocalOnlyDecider From(Func<Exception, Directive> localOnlyDecider)
+        {
+            return new LocalOnlyDecider(localOnlyDecider);
+        }
+    }
+
+    public class LocalOnlyDecider : IDecider
+    {
+        private readonly Func<Exception, Directive> _decider;
+        public LocalOnlyDecider(Func<Exception, Directive> decider)
+        {
+            _decider = decider;
+        }
+
+        public Directive Decide(Exception cause)
+        {
+            return _decider(cause);
+        }
+    }
+
+    public class DeployableDecider : IDecider
+    {
+        public DeployableDecider(Directive defaultDirective, params KeyValuePair<Type, Directive>[] pairs)
+        {
+            DefaultDirective = defaultDirective;
+            Pairs = pairs;
+        }
+
```

Comment:
I've added this to the DSL now, any reason why you would want it on the `DeployableDecider` also?